geomstats/learning/wrapped_gaussian_process.py

```diff
@@ -0,0 +1,290 @@
+                #   optional flag eval_gradient, which determines if the
+                #   gradient is returned additionally to the function value
+                # * 'initial_theta': the initial value for theta, which can be
+                #   used by local optimizers
+                # * 'bounds': the bounds on the values of theta
+                ....
+                # Returned are the best found hyperparameters theta and
+                # the corresponding value of the target function.
+                return theta_opt, func_min
+        Per default, the L-BFGS-B algorithm from `scipy.optimize.minimize`
+        is used. If None is passed, the kernel's parameters are kept fixed.
+        Available internal optimizers are: `{'fmin_l_bfgs_b'}`.
+    n_restarts_optimizer : int, default=0
+        The number of restarts of the optimizer for finding the kernel's
+        parameters which maximize the log-marginal likelihood. The first run
+        of the optimizer is performed from the kernel's initial parameters,
+        the remaining ones (if any) from thetas sampled log-uniform randomly
+        from the space of allowed theta-values. If greater than 0, all bounds
+        must be finite. Note that `n_restarts_optimizer == 0` implies that one
+        run is performed.
+    copy_X_train : bool, default=True
+        If True, a persistent copy of the training data is stored in the
+        object. Otherwise, just a reference to the training data is stored,
+        which might cause predictions to change if the data is modified
+        externally.
+    random_state : int, RandomState instance or None, default=None
+        Determines random number generation used to initialize the centers.
+        Pass an int for reproducible results across multiple function calls.
+
+    [1] Mallasto, A. and Feragen, A. Wrapped gaussian process
+    regression on riemannian manifolds. In 2018 IEEE/CVF
+    Conference on Computer Vision and Pattern Recognition
+    """
+
+    def __init__(
+        self,
+        space,
+        metric,
+        prior,
+        kernel=None,
+        *,
+        alpha=1e-10,
+        optimizer="fmin_l_bfgs_b",
+        n_restarts_optimizer=0,
+        copy_X_train=True,
+        random_state=None,
+    ):
+        if metric is None:
+            metric = space.metric
+        self.metric = metric
+        self.space = space
+        self.prior = prior
+        self.copy_X_train = copy_X_train
+
+        self.y_train_ = None
+        self.tangent_y_train_ = None
+
+        self._euclidean_gpr = GaussianProcessRegressor(
+            kernel=kernel,
+            alpha=alpha,
+            optimizer=optimizer,
+            n_restarts_optimizer=n_restarts_optimizer,
+            normalize_y=False,
+            copy_X_train=copy_X_train,
+            random_state=random_state,
+        )
+
+        # I know that inheritance seems more appropriate here, but the issue is
```

Comment:
Sounds good! Could you just remove this comment?

Reply:
done